doc/ug/dv_methodology/index.md

```diff
@@ -299,60 +299,112 @@ Once the regression tool is developed, it will provide a way to enable these cap
+These constructs also allow us to capture whether multiple scenarios have occurred simultaneously through crosses.
+This is also often referred to as cross coverage.
+These constructs are typically encapsulated in a class and are sometimes referred to as the 'functional coverage model'.
+They are sampled in 'reactive' components of the testbench, such as monitors and/or the scoreboards.
 They can also be embedded within the RTL to sample interesting scenarios through DV stimulus.
 
 Here are the metrics used with a brief explanation:
 
-*  **Assert Coverage**: This metric measures which assertions, cover properties and sequences have been exercised in the course of the simulation.
+* **Assertion Coverage using SVA**: This metric measures which assertions have been exercised in the course of the simulation.
+SVA coverage is often implemented by Design engineer and is used to add cover procedures and variables not visible on output pins.
   Note, an assertion precondition counts as a cover point.
-*  **Covergroup Coverage**: This metric (sometimes called "Testbench Coverage") measures covergroups during simulation.
-  These are usually coded in the testbench to check that the desired stimulus was generated properly, but can also be embedded in the RTL.
+* **Covergroup Coverage**: Observes values on buses, registers and so on.
+  This can verify that a specific value or range of values was seen on a bus and that no illegal values were seen.
+  Simulators can be set to throw an error if an illegal value is seen.
+  Another example of a covergroup could be the intersection of two cover points.
+  One example could be observing the FIFO full signal and the FIFO write enable signal is asserted at the same time.
+  This is called a coverage cross.
+* **Cover Property coverage**: This observes events or series of events.
+  As an example think of the TL UL register bus used in OpenTitan.
+  A cover property cover point could be the handshaking between valid and ready.
+  There is a big overlap in what can be covered by SVA and cover properties.
+  Suggestions on when to use which form of coverage are presented later in this document.
+
+
+#### Do we need both types of coverage
+
+Reaching a 100% code coverage does not tell you the whole story.
+Think of the simple example of an AND gate with inputs A, B, and output X.
+To get to 100 percent code coverage only two different input combinations are needed.
+00 and 11, these two will produce all possible outcomes of X.
+
+![single AND gate](dv_method_single_gate.svg)
+
+
+The coverage will indicate that all code was exercised.
+But we do not know that our design works as intended.
+All we know is that A, B, and X have been observed to take on both logic 0 and 1.
+We could not say for certain that the design was in fact an AND gate it could just as easily be an OR gate.
+So we need functional coverage to tell us this.
```

Comment:
Are we saying we want functional coverage to be implemented mandatorily regardless of how we are testing it?
Seems like a directed test (as opposed to constrained-random) that checks all combinations of A and B against the output Y can prove that it is indeed an AND gate without needing functional coverage to be instrumented. 

Functional coverage is no doubt, a concrete proof that we did verify a particular design functionality. However, at the chip level for example, we have almost exclusively all SW based directed tests. Schedule-wise it is a prohibitive to write functional coverage for all of those tests to prove that we did verify what we expected to verify. So we instead rely on 'testplan coverage' (all tests indicated in our testplan are passing in the regressions). Let me know your thoughts on this.

Reply:
From my point of view, there are two questions:

1. Is this enough to sign off the chip?
2. Is this enough to be absolutely sure we've got it right?

If our signoff process isn't silly, we should definitely have 2 => 1, but it seems reasonable to tape out a chip when you're just pretty confident that it's correct (the exact value of "pretty" here depends on schedule pressure from the marketing department, your personal level of risk aversion, the company budget etc.!)

I think Rasmus is right in saying that these functional coverage points are required if you want to get 2. Otherwise, how are we "testing the tests"? If I'm trying to exercise something non-trivial, I can write a UVM sequence that I think should get me in a corner. I can even look at some waves to check I got it right. But what happens if the design is modified? Do I have to recheck the waves for each of my sequences? (clearly, this isn't going to happen!). I guess that functional coverage collection is a way to automate the "looks good to me" check that the DV engineer does while writing the tests in the first place.

Whether they are required for 1 is a judgement call, I guess. Personally, I'd try to do them, just to avoid having to worry near the tapeout deadline.

Reply:
@sriyerg @rswarbrick 
As Rupert says, I am trying to convey the message - if you don't have functional coverage how do you know you when you are done testing - how do you know the tests were good enough.
seems a bit arbitrary to me to say we are done when all of the tests we could think if making are passing.

I think we are hitting a cost vs timeline common issue.
In my experience it is always cheaper to go the extra mile and implement the functional coverage rather than relying upon directed tests as there are always corners that a missed.
and usually every respin means an additional million USD in cost and a 3-6 month schedule hit.

so unless you are depending on making the date of a specific shuttle run I would say - do functional testing.

there should be some value in chaining the IP environments.

Reply:
You know when you are done testing through completion of the well reviewed testplan (i.e. testplan annotated regression result hitting 100%) and the integration code coverage closure. For SoC level integration testing (with already pre-verified IPs), this is often more than sufficient.

I am fine with the wording - I will just update the chip DV doc to say that we are waiving off the requirement to develop the functional coverage model.